client/test/client.go

```diff
@@ -85,6 +90,55 @@ func (c *Client) Deposit(ctx context.Context, amount *big.Int) error {
 	return nil
 }
 
+func (c *Client) Exit(ctx context.Context, bal *tee.BalanceProof) error {
+	tr, err := c.ethClient.NewTransactor(ctx, big.NewInt(0), eth.DefaultGasLimit, c.ethClient.Account())
+	if err != nil {
+		return fmt.Errorf("creating transactor: %w", err)
+	}
+
+	tx, err := c.contract.Exit(tr, bindings.ErdstallBalance{
+		Epoch:   bal.Balance.Epoch,
+		Account: bal.Balance.Account,
+		Value:   bal.Balance.Value,
+	}, bal.Sig)
```

Comment:
You can avoid all code duplication between `Deposit`, `Exit` and `Withdraw` if you create a method `call(ctx context.Context, call func(ctx) (types.Transaction, error)` that contains the same body as this function but here just calls `tx, err := call(ctx)` and then use it to define `Deposit`, `Exit` and `Withdraw`.

This way we'd also fix a "bug" in `Deposit` that we don't check for a successful tx execution right now :)